bpf/bpf_lxc.c

```diff
@@ -1803,6 +1804,19 @@ ipv4_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label,
 	 * want to execute the conntrack logic so that replies can be correctly
 	 * matched.
 	 */
+
+	/* If ip4.saddr is IPV4_LOOPBACK, this is almost certainly a loopback
+	 * connection. Populate
+	 * - .loopback, so that policy enforcement is bypassed, and
+	 * - .rev_nat_index, so that replies can be RevNATed.
+	 */
+	if (ret == CT_NEW && ip4->saddr == IPV4_LOOPBACK &&
+	    ct_has_loopback_egress_entry4(get_ct_map4(tuple), tuple,
+					  &ct_state_new.rev_nat_index)) {
+		ct_state_new.loopback = true;
+		goto skip_policy_enforcement;
```

Comment:
we have another issue with hairpin trafafic and network policies , @sypakine  knows more about it

Reply:
I'll file a new issue for it.

Reply:
yeah, please do, we should not conflate both problems, Thanks

Reply:
filed https://github.com/cilium/cilium/issues/27709